custom_components/sonnenbatterie/sensor.py

```diff
@@ -110,7 +111,7 @@ def set_state(self, state):
             return
         self._state = state
         if self.hass is None:
-            LOGGER.warning("hass not set, sensor: {} ".format(self.name))
+            #LOGGER.warning("hass not set, sensor: {} ".format(self.name))
```

Comment:
it should spam the log once or twice.. but not after..
did this happen to only one specifice sensor or to all?

Reply:
I can't tell exactly because it was just... flooded. HA reported about 45k entries in like 4 hours. But it showed me about 10 entities if I remember correctly. If you want, I can check again.

Reply:
Yeah i am okay with disabling this log.. but if the hass variable is not set, the sensor wont be updated.. so there might be more to this for you.

Reply:
My sensors are updating perfectly fine so I also don't really know why that happens. I'll need to investigate the code further and try to adjust it to the HA standards (if I understand how it works and get it working :smile:, so far I'm also no expert in python but have relatively much experience with it.).